### PR TITLE
support changing rate limiter dynamically

### PR DESCRIPTION
**Pull request by rleungx**
This PR is going to support changing the rate limiter dynamically.

### CODE REVIEW

File: librocksdb_sys/src/lib.rs [hunk 1]
```diff
@@ -734,6 +734,10 @@ extern "C" {
     pub fn crocksdb_options_set_delayed_write_rate(options: *mut Options, rate: u64);
     pub fn crocksdb_options_set_force_consistency_checks(options: *mut Options, v: bool);
     pub fn crocksdb_options_set_ratelimiter(options: *mut Options, limiter: *mut DBRateLimiter);
+    pub fn crocksdb_options_set_ratelimiter_bytes_per_second(
```
Comment (yiwu-arbug) [inline librocksdb_sys/src/lib.rs:737]: Can we split this method into a `crocksdb_options_ratelimiter()` getter and a `crocksdb_ratelimiter_set_bytes_per_second()` function instead?

File: src/rocksdb_options.rs [hunk 2]
```diff
@@ -1006,6 +1006,35 @@ impl DBOptions {
...
+            crocksdb_ffi::crocksdb_ratelimiter_set_bytes_per_second(limiter, rate_bytes_per_sec);
+        }
+        Ok(())
+    }
+
+    pub fn get_rate_bytes_per_sec(
+        &mut self,
+    ) -> Result<i64, String> {
+        let limiter = unsafe { crocksdb_ffi::crocksdb_options_get_ratelimiter(self.inner) };
+        if limiter.is_null() {
```
Comment (yiwu-arbug) [inline src/rocksdb_options.rs:1028]: how about returning Option<i64> instead and return None for the null case?

**Comment by rleungx** [comment]
PTAL @yiwu-arbug @Connor1996

File: librocksdb_sys/crocksdb/c.cc [hunk 3]
```diff
@@ -2907,6 +2907,16 @@ void crocksdb_options_set_ratelimiter(crocksdb_options_t* opt,
   limiter->rep = nullptr;
 }
 
+crocksdb_ratelimiter_t* crocksdb_options_get_ratelimiter(
+    crocksdb_options_t* opt) {
+  if (opt->rep.rate_limiter != nullptr) {
+    crocksdb_ratelimiter_t* limiter = new crocksdb_ratelimiter_t;
```
Comment (Connor1996) [inline librocksdb_sys/crocksdb/c.cc:2913]: wait, seems here is a memory leak

**Pull request merged**